### PR TITLE
Fixes for wx savefig dialog.

### PR DESCRIPTION
**Pull request by anntzer**
1) GetDirectory() returns the currently selected directory, which may
   not match the user-input directory, e.g. if the user types in an
   absolute path ("/home/user/foo.png") -- in that case the file was
   saved to the wrong directory.  Instead use GetPath(), which returns
   the full path.
2) Correctly use and update rcParams["savefig.directory"], as in the
   other backends.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by anntzer** [comment]
I have tested this; GetPath() returns the path as it is input, regardless of the filetype you pick in the dropdown widget -- but the dropdown widget sets the default extension in the path input field.  So if you pick ".png" in the dropdown widget the filename defaults to foo.png, but you *can* edit it to be "foo.baz" instead and then your filename will be "foo.baz".  That's essentially what the issue you linked says, no?

**Comment by timhoffm** [comment]
The issue says

> if my wildcard is set to *.txt and I go to /tmp and type “test” in the filename box sand click save, GetPath() returns “/tmp/test” instead of “/tmp/test.txt”.

but you use `path.suffix[1:]`, which accroding to the issue is not guaranteed to exist.

**Comment by anntzer** [comment]
Then ext is just the empty string and the warning does not occur.

The point of the warning is "if I type foo.pdf but the wildcard is set to png, save to pdf despite the wildcard".  If I type just foo without extension and having manually deleted the extension from the input dialog, then we use whatever the wildcard dropbox is set to.

File: lib/matplotlib/backends/backend_wx.py [hunk 1]
```diff
@@ -1374,29 +1374,28 @@ def save_figure(self, *args):
...
                 # looks like they forgot to set the image type drop
                 # down, going with the extension.
                 _log.warning('extension %s did not match the selected '
                              'image type %s; going with %s',
-                             ext, format, ext)
-                format = ext
+                             ext, fmt, ext)
+                fmt = ext
+            # Save dir for next time, unless empty str (which means use cwd).
+            if rcParams["savefig.directory"]:
```
Comment (jklymak) [inline lib/matplotlib/backends/backend_wx.py]: You seem to have a different philosophy abut `rcParams` than I do.  I think of it as defaults the user/matplotlib have set, not something we change as the GUI gets run.  i.e. if I set `save fig.directory` in my matplotlibrc, I don't expect it to change?

Comment (timhoffm) [reply lib/matplotlib/backends/backend_wx.py]: That has been the way in all backends. From a users point of view, remembering the last used directory in a save dialog is the expected (=standard) behavior for an application.

One can argue if that's should still be the case between multiple runs of the program. But that would be a change with respect to the current behavior.

Comment (jklymak) [reply lib/matplotlib/backends/backend_wx.py]: Oh duh, sorry didn't notice that it said the other backends do that.  Sorry.

**Pull request merged**